### PR TITLE
feat: conditional formatting multiple rules

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/3932

### Description:

- [x] I am able to add as many rules as I want to my table conditional formatting (should we set some limit, like 20? Or is there a point where the number of rules decreases the performance of the app?)
- [x] I am able to select a different colour for each of my rules
- [x] You can only use a column in one rule (you can't use the same column across many rules)

https://user-images.githubusercontent.com/962095/212127900-76362706-c3c9-4ed5-aa06-6c6f0d535bf6.mp4

### CODE REVIEW

**Comment by ZeRego** [comment]
> You can only use a column in one rule (you can't use the same column across many rules)

Is there a ticket to allow multiple rules for the same field? or at least an if/else colour? 
I wanted to make all numbers bellow 0 red and above 0 green.

File: packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            ))}
+
+            <FormGroup>
+                <Tooltip2
+                    position="bottom-left"
+                    disabled={
+                        visibleActiveNumericFields.length !==
+                        usedFieldIds.length
+                    }
+                    content="All fields are already being used in rules."
```
Comment (ZeRego) [inline packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx:96]: Blueprint has some issues with tooltips on disabled elements. 


https://user-images.githubusercontent.com/9117144/212322848-ce1118ad-ca12-4369-9101-da6b5ebde099.mp4

File: packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    draft[index] = newConfig;
+                }),
+            ),
+        [onSetConditionalFormattings, activeConfigs],
+    );
+
+    return (
+        <ConditionalFormattingListWrapper>
+            {activeConfigs.map((conditionalFormatting, index) => (
+                <ConditionalFormatting
```
Comment (ZeRego) [inline packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx:78]: Suggestion: have the rules collapsed if there are than one

`isDefaultOpen={activeConfigs.lenght > 1}`

**Review by ZeRego — APPROVED** [review]
working well 👍 

just left 2 small suggestions

File: packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx:78]: good suggestion, I'll make a change 👍

File: packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx:96]: In `packages/frontend/src/components/RefreshButton.tsx` we only used the disabled class and prevented the onClick action.

```
                // disabled button captures hover events
                onClick={isButtonDisabled ? undefined : onClick}
                className={isButtonDisabled ? Classes.DISABLED : undefined}
            ```

Comment (IrakliJani) [reply packages/frontend/src/components/TableConfigPanel/ConditionalFormattingList.tsx:101]: that worked perfectly, thanks!

**Pull request merged**